Examples/destination_plugins/AppsFlyerDestination.swift

```diff
@@ -0,0 +1,352 @@
+    let name: String
+    var analytics: Analytics?
+    
+    internal var settings: AppsFlyerSettings? = nil
+    
+    required init(name: String) {
+        self.name = name
+        analytics?.track(name: "AppsFlyer Loaded")
+    }
+    
+    public func update(settings: Settings) {
+        
+        guard let settings: AppsFlyerSettings = settings.integrationSettings(name: "AppsFlyer") else { return }
+        self.settings = settings
+        
+        AppsFlyerLib.shared().appsFlyerDevKey = settings.appsFlyerDevKey
+        AppsFlyerLib.shared().appleAppID = settings.appleAppID
+        
+        AppsFlyerLib.shared().waitForATTUserAuthorization(timeoutInterval: 60) //OPTIONAL
+        AppsFlyerLib.shared().isDebug = true //OPTIONAL
+        AppsFlyerLib.shared().deepLinkDelegate = self //OPTIONAL
+        
+        let trackAttributionData = settings.trackAttributionData
+        
+        if trackAttributionData ?? false {
+            AppsFlyerLib.shared().delegate = self
+        }
+        
+        func applicationDidBecomeActive(application: UIApplication) {
+            AppsFlyerLib.shared().start()
+        }
+        
+        func openURL(_ url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) {
+            AppsFlyerLib.shared().handleOpen(url, options: options)
+        }
+        
+        func receivedRemoteNotification(userInfo: [AnyHashable: Any]) {
+            AppsFlyerLib.shared().handlePushNotification(userInfo)
+        }
+    }
+    
+    public func identify(event: IdentifyEvent) -> IdentifyEvent? {
+        if let userId = event.userId, userId.count > 0 {
+            AppsFlyerLib.shared().customerUserID = userId
+        }
+        
+        if let traits = event.traits?.dictionaryValue {
+            var aFTraits: [AnyHashable: Any] = [:]
+            
+            if let email = traits["email"] as? String {
+                aFTraits["email"] = email
+            }
+            
+            if let firstName = traits["firstName"] as? String {
+                aFTraits["firstName"] = firstName
+            }
+            
+            if let lastName = traits["lastName"] as? String {
+                aFTraits["lastName"] = lastName
+            }
+            
+            if traits["currencyCode"] != nil {
+                AppsFlyerLib.shared().currencyCode = traits["currencyCode"] as? String
+            }
+            
+            AppsFlyerLib.shared().customData = aFTraits
+            
+            print("hello")
+        }
+        
+        return event
+    }
+    
+    public func track(event: TrackEvent) -> TrackEvent? {
+        
+        var properties = event.properties?.dictionaryValue
+        
+        let revenue: Double? = extractRevenue(key: "revenue", from: properties)
+        let currency: String? = extractCurrency(key: "currency", from: properties, withDefault: "USD")
+        
+        if let af_revenue = revenue, let af_currency = currency {
```

Comment:
Even when our json or properties may be snake case, lets keep camelCase in actual swift